src/phase0/presets/mainnet.rs

```diff
@@ -0,0 +1,96 @@
+pub use crate::domains::{DomainType, SigningData};
```

Comment:
the general approach is to re-export items in a given preset and *this* is where we can set the appropriate constants for the compile-time data that changes across preset.

things that aren't great:
1) still have _some_ code duplication across presets... i started writing a macro to resolve this so that the preset just looks like:

```rust
define_preset! {
    MAX_COMMITTEE_PER_SLOT: u64 = 64;
    TARGET_COMMITTEE_SIZE: u64 = 128;
    ...
    MAX_VOLUNTARY_EXITS: usize = 16; 
}
```
and think this is advisable after we have sketched out more of the state transition and have a better idea of how it will all fit together (bc debugging thru the macro could be annoying)

2) the types are a bit heavy w/ 6-8 const generics that have to flow up to the top-level operators of this crate.... given the above re: Rust's current feature set, this may be the best we can do right now...